From c82095f528faf7f73d087dcd635ee8eabf164668 Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Wed, 30 Mar 2016 15:59:34 +0800
Subject: [PATCH] ppc32: kexec: only shutdown online CPUs

present CPUs may not equal online CPUs, we should power off online CPUs
rather than present CPUs.

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/powerpc/platforms/85xx/smp.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/platforms/85xx/smp.c b/arch/powerpc/platforms/85xx/smp.c
index 3511e50..b5c057d7 100644
--- a/arch/powerpc/platforms/85xx/smp.c
+++ b/arch/powerpc/platforms/85xx/smp.c
@@ -726,7 +726,7 @@ static void mpc85xx_smp_machine_kexec(struct kimage *image)
 {
 #ifdef CONFIG_PPC32
 	int timeout = INT_MAX;
-	int i, num_cpus = num_present_cpus();
+	int i, num_cpus = num_online_cpus();
 #endif
 
 #ifdef CONFIG_PPC32
-- 
1.7.5.4

